src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1011,6 +1022,17 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
                                              diagnostics);
                 }
 
+                gotError |= !CheckInvocationArgMixing(
+                            node,
+                            method,
+                            receiver,
+                            method.Parameters,
+                            args,
+                            argRefKinds,
+                            argsToParams,
+                            this.LocalScopeDepth,
+                            diagnostics);
```

Comment:
`CheckInvocationArgMixing()` call can be moved ahead of `if` and shared between both branches.